maven/lib/dependabot/maven.rb

```diff
@@ -23,6 +23,6 @@
     "maven",
     lambda { |name|
       _group_id, artifact_id, _classifier = name.split(":")
-      %w(bom library).include?(artifact_id) ? name : artifact_id
+      name.length <= 100 ? name : artifact_id
```

Comment:
I am not sure if we should. I just used an existing pull request as a model https://github.com/dependabot/dependabot-core/pull/6526

Whatever we decide to do should probably be done for both ecosystems to have consistency among the different Java package managers

Reply:
Oh, I did not realize we had already done this for Gradle! It looks good then, since nobody complained there.

For completeness, this is where the special case was added: https://github.com/dependabot/dependabot-core/pull/1203, to fix https://github.com/dependabot/feedback/issues/513.